### PR TITLE
review: test: add test for createReference

### PR DESCRIPTION
**Pull request by Rohitesh-Kumar-Jain**
Hi, I have added a new test for the method ```createRefernce``` :

This test is intended to kill these mutations:

Screenshot 2021-08-08 at 10 21 47 PM

Screenshot 2021-08-09 at 11 29 19 AM

Link to #3967

### CODE REVIEW

File: src/test/java/spoon/test/factory/MethodFactoryTest.java [hunk 1]
```diff
@@ -31,4 +32,30 @@ public void testCreateReference() {
...
+        Factory factory = new Launcher().getFactory();
+        Class<?> testClass = Class.forName("spoon.test.factory.testclasses.Bar");
+        Method testMethod = testClass.getMethod("foo");
+
+        CtExecutableReference<?> expectedReference = factory.createExecutableReference();
+        expectedReference.setSimpleName("foo");
+        CtTypeReference<?> ctTypeReference = factory.Type().createReference(Bar.class);
+        expectedReference.setDeclaringType(ctTypeReference);
+        CtTypeReference<?> typeReference = factory.Type().createReference(testMethod.getReturnType());
+//        expectedReference.setType(typeReference);
```
Comment (Rohitesh-Kumar-Jain) [inline src/test/java/spoon/test/factory/MethodFactoryTest.java:50]: I am not able to figure out how to set the type of the expected reference to "void", I need help here

Comment (slarse) [reply src/test/java/spoon/test/factory/MethodFactoryTest.java:50]: You mean something like this?

```java
expectedReference.setType(factory.Type().voidPrimitiveType());
```

Comment (Rohitesh-Kumar-Jain) [reply src/test/java/spoon/test/factory/MethodFactoryTest.java:50]: yes I meant something like this only, thank you so much

**Review by slarse — APPROVED** [review]
LGTM, thanks @Rohitesh-Kumar-Jain!

**Pull request merged**